### PR TITLE
AST-378 - fix: Related Posts QA reported improvements & fixes

### PR DESCRIPTION
**Pull request by imnavanath**
### Description
- Task - https://wp-astra.atlassian.net/browse/AST-378
- QA Reported issues

### Types of changes
- Non breaking changes

### How has this been tested?
- With tasks mentioned in above task

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/class-astra-dynamic-css.php [hunk 1]
```diff
@@ -3569,7 +3572,10 @@ public static function related_posts_static_css() {
 				width: 100%;
 				position: relative;
 			}
-			.ast-related-post-featured-section + .entry-header, .ast-related-post-featured-section + .entry-content {
+			.ast-related-posts-wrapper .ast-related-posts-inner-section {
```
Comment (premanshup) [inline inc/class-astra-dynamic-css.php:3575]:
```suggestion
			.ast-related-posts-inner-section {
```
@imnavanath I think we can reduce the CSS weightage here

File: inc/dynamic-css/related-posts.php [hunk 2]
```diff
@@ -55,9 +55,16 @@ function astra_related_posts_css( $dynamic_css ) {
 		$related_posts_section_title_line_height    = astra_get_option( 'related-posts-section-title-line-height' );
 		$related_posts_section_title_text_transform = astra_get_option( 'related-posts-section-title-text-transform' );
 
+		// Setting up container BG color by default to Related Posts's section BG color.
+		$content_bg_obj     = astra_get_option( 'content-bg-obj-responsive' );
+		$container_bg_color = '#ffffff';
+		if ( isset( $content_bg_obj['desktop'] ) && isset( $content_bg_obj['desktop']['background-color'] ) && '' !== $content_bg_obj['desktop']['background-color'] ) {
```
Comment (premanshup) [inline inc/dynamic-css/related-posts.php:61]:
```suggestion
		if ( isset( $content_bg_obj['desktop']['background-color'] ) && '' !== $content_bg_obj['desktop']['background-color'] ) {
```
Please verify this once. Direct checking isset for key to the child key should work.

**Review by premanshup — CHANGES_REQUESTED** [review]

File: inc/dynamic-css/related-posts.php (hunk 2 above)
Comment (imnavanath) [reply inc/dynamic-css/related-posts.php]: Done 👍

File: inc/class-astra-dynamic-css.php (hunk 1 above)
Comment (imnavanath) [reply inc/class-astra-dynamic-css.php]: Done 👍

**Review by premanshup — APPROVED** [review]

**Pull request merged**